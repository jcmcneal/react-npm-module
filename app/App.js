import React from 'react';
import ReactDOM from 'react-dom';
import { hot } from 'react-hot-loader';

import Component from '../dist/bundle.min';

const App = () => (
    <div>
        This is the sandbox app to test your npm module.
        <Component />
    </div>
);

/** HMR */
hot(module)(App);

ReactDOM.render(<App />, document.getElementById('app'));
