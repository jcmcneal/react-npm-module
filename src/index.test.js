import React from 'react';
import Renderer from 'react-test-renderer';

import App from './index';

/** Helper Functions */
const render = jsx => Renderer.create(jsx);
const getInstance = (component) => {
    const wrapper = render(component);
    return wrapper.getInstance();
};
const getState = (component) => {
    const instance = getInstance(component);
    return instance ? instance.state : undefined;
};
const takeSnapshot = test => expect(test).toMatchSnapshot();
const getTreeSnapshot = component => takeSnapshot(render(component));
const testComponent = (component) => {
    getTreeSnapshot(component);

    const state = getState(component);
    if (state) takeSnapshot(state);

    return state;
};

/** Unit Tests */
describe('Component', () => {
    it('basic snapshot', () => {
        testComponent(<App />);
    });
});
