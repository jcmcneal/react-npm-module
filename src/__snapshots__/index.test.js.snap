// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component basic snapshot 1`] = `
<div>
  Hello Worlds.
</div>
`;
